tests/rptest/scale_tests/tiered_storage_stability_test.py

```diff
@@ -393,3 +399,159 @@ def stage_cloud_cache_thrash(self):
+
+        wait_until(producer_complete,
+                   timeout_sec=required_wait_time_s,
+                   backoff_sec=30)
+
+        post_prod_offsets = [(p.id, p.high_watermark)
+                             for p in self.rpk.describe_topic(self.topic_name)
+                             if p.high_watermark is not None]
+
+        offset_deltas = [
+            (c[0][0], c[0][1] - c[1][1])
+            for c in list(itertools.product(post_prod_offsets, last_offsets))
+            if c[0][0] == c[1][0]
+        ]
+        avg_delta = sum([d[1] for d in offset_deltas]) / len(offset_deltas)
+
+        self.logger.info(
+            f"Finished waiting for batch cache to fill. Avg log offset delta: {avg_delta}"
+        )
+
+        def check_partition_size(old_size, new_size):
+            self.logger.info(
+                f"Total increase in size for partitions: {new_size-old_size}")
+
+            unreplicated_size_inc = (new_size - old_size) / 3
+            return unreplicated_size_inc >= batch_cache_max_memory
+
+        # Ensure total partition size increased by the amount expected.
+        for check in partition_size_check:
+            check.expect([(partition_size_metric, check_partition_size)])
+
+        # Ensure there are metrics available for the topic we're consuming from
+        for p_id, _ in last_offsets:
+            self._consume_from_offset(self.topic_name, 1, p_id, "newest", 10)
+
+        # Stop the producer temporarily as produce requests can cause
+        # batch cache reads which causes false negatives on cache misses.
+        producer.stop()
+
+        check_batch_cache_reads = []
+        cache_metrics = [
+            "vectorized_storage_log_read_bytes_total",
+            "vectorized_storage_log_cached_read_bytes_total",
+        ]
+
+        for node in self.redpanda.nodes:
+            check_batch_cache_reads.append(
+                MetricCheck(self.logger,
+                            self.redpanda,
+                            node,
+                            cache_metrics,
+                            reduce=sum))
+
+        # start consuming at the offsets recorded before the wait.
+        # at this point we can be sure they are not from the batch cache.
+        messages_to_read = 1
+        timeout_seconds = 60
+
+        for p_id, p_hw in last_offsets:
+            self._consume_from_offset(self.topic_name, messages_to_read, p_id,
+                                      str(p_hw), timeout_seconds)
+
+        def check_cache_bytes_ratio(old, new):
+            log_diff = int(new[cache_metrics[0]]) - int(old[cache_metrics[0]])
+            cache_diff = int(new[cache_metrics[1]]) - int(
+                old[cache_metrics[1]])
+            cache_hit_percent = cache_diff / log_diff
+
+            self.logger.info(
+                f"BYTES: log_diff: {log_diff} cache_diff: {cache_diff} cache_hit_percent: {cache_hit_percent}"
+            )
+            return cache_hit_percent <= 0.4
```

Comment:
Curious in practice what this tends to be?

Reply:
cc @ballard26